pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -73,11 +73,13 @@ func checkProxyDependencies(proxy *model.Proxy, config model.ConfigKey, push *mo
 			}
 
 			// If config is ServiceEntry, name of the config is service's FQDN
-			svc, exist := push.ServiceIndex.HostnameAndNamespace[host.Name(config.Name)][config.Namespace]
-			if exist {
-				if !push.IsServiceVisible(svc, proxy.Metadata.Namespace) {
-					return false
-				}
+			hostname := host.Name(config.Name)
+			// gateways have default sidecar scopes
+			if proxy.SidecarScope.GetService(hostname) == nil &&
+				proxy.PrevSidecarScope.GetService(hostname) == nil {
```

Comment:
proxy.PrevSidecarScope can be nil

Reply:
I've already checked it
https://github.com/istio/istio/pull/42422/files#diff-42d26d46fa9490c70f6bd91080addf3ecc57aa44ebae320708ae58c824af8a9dR553

Reply:
lg